cni/network/invoker_cns.go

```diff
@@ -164,20 +163,27 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 			zap.Any("podInfo", podInfo))
 
 		//nolint:exhaustive // ignore exhaustive types check
+		// Do we want to leverage this lint skip in other places of our code?
 		switch info.nicType {
 		case cns.DelegatedVMNIC:
 			// only handling single v4 PodIPInfo for DelegatedVMNICs at the moment, will have to update once v6 gets added
 			if !info.skipDefaultRoutes {
 				numInterfacesWithDefaultRoutes++
 			}
 
+			// Add secondary interface info from podIPInfo to ipamAddResult
+			info.hostSubnet = response.PodIPInfo[i].HostPrimaryIPInfo.Subnet
+			info.hostPrimaryIP = response.PodIPInfo[i].HostPrimaryIPInfo.PrimaryIP
+			info.hostGateway = response.PodIPInfo[i].HostPrimaryIPInfo.Gateway
+
 			if err := configureSecondaryAddResult(&info, &addResult, &response.PodIPInfo[i].PodIPConfig); err != nil {
 				return IPAMAddResult{}, err
 			}
-		default:
+		case cns.InfraNIC:
```

Comment:
my code is expecting that once ipamAddResult and interfaceInfo inside it is populated and returned, there should always be a nic type-- if it is empty it should be populated with something, likely infra nic if nothing matches-- it will break if the nic type is left empty. To address your comment, if the nic type is empty we should perform the same actions as if it were "infra" (and populate the nic type as infra)-- will also check invoker_azure and multitenancy.go to check cases where nic type is empty

Reply:
confirmed in multitenancy.go and invoker_azure.go that we always set nic type to infra when adding to the interface info map